drivers/i3c/Kconfig

```diff
@@ -0,0 +1,93 @@
+# I3C configuration options
+#
+# Copyright (c) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig I3C
+	bool "I3C Drivers"
+	help
+	  Enable I3C Driver Configuration
+
+if I3C
+
+module = I3C
+module-str = i3c
+source "subsys/logging/Kconfig.template.log_config"
+
+config I3C_USE_GROUP_ADDR
+	bool "Use Group Addresses"
+	default y
+	help
+	  Enable this to use group addresses if supported
+	  by the controllers and target devices.
+
+	  Says Y if unsure.
+
+config I3C_USE_IBI
+	bool "Use In-Band Interrupt (IBI)"
+	default y
+	help
+	  Enable this to use In-Band Interrupt (IBI).
+
+	  Says Y if unsure.
+
+config I3C_IBI_MAX_PAYLOAD_SIZE
+	int "Maximum IBI Payload Size"
+	depends on I3C_USE_IBI
+	default 16
+	help
+	  Maxmium IBI payload size.
+
+comment "Initialization Priority"
+
+config I3C_CONTROLLER_INIT_PRIORITY
+	int "I3C Controller Init Priority"
+	# Default is just after CONFIG_KERNEL_INIT_PRIORITY_DEVICE
+	default 50
+	help
+	  This is for setting up I3C controller device driver instance
+	  so device driver instances for I2C and I3C devices on
+	  the same bus can register themselves with the I3C controller
+	  instance.
+
+	  Device driver may initialize the hardware at this point, or
+	  choose to do so during bus initialization.
+
+config I3C_DEV_REGISTER_INIT_PRIORITY
+	int "I3C Device Registration Init Priority"
+	default 60
```

Comment:
Because of the usage of `struct i3c_i2c_device_desc`, each device driver will need to do its own magic to register themselves to the I3C controller, as each driver needs to allocate space for the struct.

As for using the struct for I2C devices, this is deliberate to align with the I3C counterpart (with `struct i3c_target_desc`) so that they look consistent.

Reply:
So what you're saying is that each and every existing i2c device driver within zephyr will need to be modified... I don't think that's feasible....
But since all i2c devices on an i3c bus should be known at compile time when the devicetree is read... the data could be read with that `I3C_I2C_DEVICE_DESC_DT(node_id)` macro you've got there and then passed into that `i3c_i2c_device_register` function.